src/Geometry.cc

```diff
@@ -75,57 +80,8 @@ Geometry::~Geometry()
 
 //////////////////////////////////////////////////
 Geometry::Geometry(const Geometry &_geometry)
-  : dataPtr(new GeometryPrivate)
+  : dataPtr(new GeometryPrivate(*_geometry.dataPtr))
```

Comment:
Out of curiosity, why aren't all these private pointers `std::unique_ptr`? It would let you have have implicit destruction, which can be more efficient.

Reply:
We are soon changing them all to `ignition::utils::ImplPtr` as started in #447. However, to answer your question, I think old code used raw pointers and was never updated. There's also a pesky warning when you use `std::unique_ptr` in an exported class in Windows.